src/gen9_avc_encoder.c

```diff
@@ -0,0 +1,6027 @@
+            75
+    },
+
+    // unsigned int 12
+    {
+            97,
+            103,
+            125,
+            160
+    },
+
+    // unsigned int 13
+    {
+            -3,
+            -2,
+            -1,
+            0
+    },
+
+    // unsigned int 14
+    {
+            1,
+            2,
+            3,
+            0xff
+    },
+
+    // unsigned int 15
+    {
+            0,
+            0,
+            0,
+            0
+    },
+
+    // unsigned int 16
+    {
+            0
+    },
+
+    // unsigned int 17
+    {
+            0
+    },
+
+    // unsigned int 18
+    {
+            0
+    },
+
+    // unsigned int 19
+    {
+            0
+    },
+
+    // unsigned int 20
+    {
+            0
+    },
+
+    // unsigned int 21
+    {
+            0
+    },
+
+    // unsigned int 22
+    {
+            0
+    },
+
+    // unsigned int 23
+    {
+            0
+    },
+
+};
+
+static void
+gen9_avc_update_misc_parameters(VADriverContextP ctx,
+                                struct encode_state *encode_state,
+                                struct intel_encoder_context *encoder_context)
+{
+    struct encoder_vme_mfc_context * vme_context = (struct encoder_vme_mfc_context *)encoder_context->vme_context;
+    struct generic_enc_codec_state * generic_state = (struct generic_enc_codec_state * )vme_context->generic_enc_state;
+    int i;
+
+    /* brc */
+    generic_state->max_bit_rate = ALIGN(encoder_context->brc.bits_per_second[0], 1000) / 1000;
+    generic_state->window_size = encoder_context->brc.window_size;
+    generic_state->brc_need_reset = encoder_context->brc.need_reset;
+
+    if (generic_state->internal_rate_mode == VA_RC_CBR) {
+        generic_state->min_bit_rate = generic_state->max_bit_rate;
+        generic_state->mb_brc_enabled = encoder_context->brc.mb_rate_control[0];
+
+        if (generic_state->target_bit_rate != generic_state->max_bit_rate) {
+            generic_state->target_bit_rate = generic_state->max_bit_rate;
+            generic_state->brc_need_reset = 1;
+        }
+    } else if (generic_state->internal_rate_mode == VA_RC_VBR) {
+        generic_state->min_bit_rate = generic_state->max_bit_rate * (2 * encoder_context->brc.target_percentage[0] - 100) / 100;
+        generic_state->mb_brc_enabled = encoder_context->brc.mb_rate_control[0];
+
+        if (generic_state->target_bit_rate != generic_state->max_bit_rate * encoder_context->brc.target_percentage[0] / 100) {
+            generic_state->target_bit_rate = generic_state->max_bit_rate * encoder_context->brc.target_percentage[0] / 100;
+            generic_state->brc_need_reset = 1;
+        }
+    }
+
+    /*  frame rate */
+    if (generic_state->internal_rate_mode != VA_RC_CQP)
+    {
+        generic_state->frames_per_100s = encoder_context->brc.framerate[0].num/encoder_context->brc.framerate[0].den * 100;
+        generic_state->frame_rate = encoder_context->brc.framerate[0].num/encoder_context->brc.framerate[0].den ;
+    }
```

Comment:
As it is pointed in previous review cycle, the above code has the precision loss.
Had better be changed to:
    num * 100 / den.